seabird_chat_ingest.proto

```diff
@@ -0,0 +1,164 @@
+syntax = "proto3";
+
+option go_package = ".;pb";
+
+import "common.proto";
+
+package seabird;
+
+message ChatUser {
+  string id = 1;
+  string name = 2;
+}
+
+message ChatChannel {
+  string id = 1;
+  string name = 2;
+}
+
+// When a chat service receives a PingChatRequest, it MUST respond with a
+// PongChatEvent containing the copied nonce.
+message PingChatRequest {
+  string nonce = 1;
+}
+
+// PongChatRequest will be sent in response to a PingChatEvent.
+message PongChatRequest {
+  string nonce = 1;
+}
+
+message SendMessageChatRequest {
+  string channel_id = 1;
+  string message = 2;
+}
+
+message SendPrivateMessageChatRequest {
+  string user_id = 1;
+  string message = 2;
+}
+
+message JoinChannelChatRequest {
+  string channel_name = 1;
+}
+
+message LeaveChannelChatRequest {
+  string channel_id = 1;
+}
+
+message SetChannelTopicChatRequest {
+  string channel = 1;
+  string topic = 2;
+}
+
+// Each ChatRequest has an ID (which can be attached to outgoing events - only
+// the first outgoing event with this ID will be used as the reply) and an inner
+// message type. Having an ID allows us to route an event back to the requestor
+// without having the chat backend implement a service.
+message ChatRequest {
+  string id = 1;
+
+  oneof inner {
+    PingChatRequest ping = 2;
+    PongChatRequest pong = 3;
+    SendMessageChatRequest send_message = 4;
+    SendPrivateMessageChatRequest send_private_message = 5;
+    JoinChannelChatRequest join_channel = 6;
+    LeaveChannelChatRequest leave_channel = 7;
+    SetChannelTopicChatRequest set_channel_topic = 8;
+  }
+}
+
+// HelloChatEvent must be the first event sent by the chat backend. If a different
+// event is sent, the connection will be closed.
+message HelloChatEvent {
+  common.Identity identity = 1;
+}
+
+// PingChatEvent is an optional event which may be sent by the chat service to
+// test latency to seabird-core. When a ping event is sent, seabird-core will
+// respond with a PongChatRequest. The nonce sent with the ping will be returned
+// in the pong.
+message PingChatEvent {
+  string nonce = 1;
+}
+
+// PongChatEvent is a required event which must be sent by the chat service
+// after receiving a PingChatRequest from seabird-core. This is used by
+// seabird-core to test latency to chat services.
+//
+// Note that if a chat backend connects to a service which implements a
+// PING/PONG system similar to seabird, it may be worthwhile to test the full
+// round trip or at least check the latency there as well.
+message PongChatEvent {
+  string nonce = 1;
+}
```

Comment:
Why make these into a stream of messages? If you're testing latency with these, the way this is being proposed tests both the latency of processing all messages, the connection latency, the overall RPC queue latency, and other stuff. I'd instead make this raw RPCs to core to eliminate the "hey I'm also working through other chat messages" latency.

Also, minor nit: the way this is named it seems like you're pinging the chat, when in fact it's the other way around. I understand that that's an effect of the `___ChatEvent` naming scheme, but I think it'd still be better to name this more accurately (and move it out of a message queue).

Reply:
This was mostly copied from how IRC handles PING/PONG as I'm most familiar with that. The idea is that it's a full round trip - in theory, it's a baseline of how long it can take for a backend to respond to a message. For now I think I'll drop this as it's been shown that it's not as valuable as I'd thought, but I'd be curious to see what a better implementation would look like without having the chat backend implement an RPC service.

Reply:
Ping/pong events were dropped for now so I'm resolving this thread.